### PR TITLE
Mailet javadocs

### PR DESCRIPTION
**Pull request by amichair**
Some Mailet API javadoc cleanup. Note that there's still work to do there - the new classes have no docs at all, and some of the old descriptions can be improved. But this is a good start :-)

### CODE REVIEW

File: mailet/api/src/main/java/org/apache/mailet/Mail.java [hunk 1]
```diff
@@ -354,18 +357,23 @@ default boolean hasSender() {
     Optional<Attribute> setAttribute(Attribute attribute);
 
     /**
-     * Store a header (and its specific values) for a recipient
-     * This header will be stored only for this recipient at delivery time
-     * 
+     * Adds a header (and its specific values) for a recipient.
+     * This header will be stored only for this recipient at delivery time.
+     * <p>
```
Comment (chibenwa) [inline mailet/api/src/main/java/org/apache/mailet/Mail.java]: Where is this closed?

**Comment by amichair** [comment]
That's the javadoc convention for separating paragraphs... no closing
tag. e.g.
https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format
(see notes) or
http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/tip/src/share/classes/java/lang/Object.java#l130
or almost any other well-formatted javadocs I've encountered :-)

**Comment by chibenwa** [comment]
Thanks for the explanation!

**Comment by amichair** [comment]
Nope, but I opened a separate JIRA issue about the deprecations, this PR is mostly about cleaning up little javadoc inconsistencies.

**Comment by ieugen** [comment]
Thank you for your contribution.
I'm looking to do a Marie Kondo on the PR's and clean them up.

I know it may be much to ask. 
If you can fix the conflicts I will make sure to merge this since it's already approved.

**Comment by ieugen** [comment]
I merged the contributions outside of this PR because of conflicts.

https://github.com/apache/james-project/commit/97c391d8e0dae062d440280300c400f6a5753cda

**Pull request closed without merging**